scrapy/core/downloader/handlers/__init__.py

```diff
@@ -20,25 +22,40 @@ def __init__(self, crawler):
             # component (extension, middleware, etc).
             if clspath is None:
                 continue
-            cls = load_object(clspath)
-            try:
-                dh = cls(crawler.settings)
-            except NotConfigured as ex:
-                self._notconfigured[scheme] = str(ex)
-            else:
-                self._handlers[scheme] = dh
+            self._schemes[scheme] = clspath
 
         crawler.signals.connect(self._close, signals.engine_stopped)
 
+    def _get_handler(self, scheme):
+        """Lazy-load the downloadhandler for a scheme
+        only on the first request for that scheme.
+        """
+        if scheme in self._handlers:
+            return self._handlers[scheme]
+        if scheme in self._notconfigured:
+            return None
+        if scheme not in self._schemes:
+            self._notconfigured[scheme] = \
+                    'no handler available for that scheme'
+            return None
+
+        dhcls = load_object(self._schemes[scheme])
```

Comment:
honestly, I am not sure it adds any value lazy loading handlers, I understand your motivation is to avoid the import of boto and related libs for spiders that doesn't use those libraries. 

I haven't an strong position, in general terms start time is proportional to the number of modules imported, so less imports means less waiting to start crawling.

In the other side, if a spider relies on a scheme that can fail, it probably prefers to know before starting to crawl. But it is also true that right now a spider can rely on a scheme that is not configured and the crawl is not stopped on start.

So, my conclusion is delaying scheme errors because of lazy-load is not a regression and the change can help reduce start time.